src/scripts/main.js

```
@@ -1,5 +1,33 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+wall.addEventListener('click', function (e) {
+  const wallRect = wall.getBoundingClientRect();
+  const BORDER_WIDTH = 10;
+  const FULL_COMPENSATION = 20;
```

These hardcoded values (`BORDER_WIDTH` and `FULL_COMPENSATION`) violate the requirement that the code should work with walls of any size and style. If the wall's border width in the CSS were changed, this logic would fail. You should retrieve the border width dynamically from the element's computed styles to make your solution more robust.